apis/elbv2/v1beta1/ingressclassparams_types.go

```diff
@@ -41,6 +41,22 @@ const (
 	LoadBalancerSchemeInternetFacing LoadBalancerScheme = "internet-facing"
 )
 
+// SubnetSelector selects one or more existing subnets.
+type SubnetSelector struct {
+	// ID specifies the resource ID of a subnet. Exactly one of this or `tagKey` must be specified.
+	// +kubebuilder:validation:Pattern=subnet-[0-9a-f]+
+	// +optional
+	ID string `json:"id,omitempty"`
```

Comment:
Why not a list of IDs or a list of tag key/values? We can limit one set of subnet selectors for the ingress group.

Reply:
Is there a need to prohibit specifying both IDs and tag/values? I'd like the interface to be able to specify anything that the existing annotation can.

Reply:
The current proposal cannot represent the rules for subnet discovery: It cannot specify that two keys must match. If we go with the `TagSelector` approach I mentioned above, then one could do:

```yaml
spec:
    subnets:
    - tags:
      - key: kubernetes.io/role/elb
        value: "1"
      - key: kubernetes.io/cluster/the-cluster-name
        value: owned
    - tags:
      - key: kubernetes.io/role/elb
        value: "1"
      - key: kubernetes.io/cluster/the-cluster-name
        value: shared
    - tags:
      - key: kubernetes.io/role/elb
        value: ""
      - key: kubernetes.io/cluster/the-cluster-name
        value: owned
    - tags:
      - key: kubernetes.io/role/elb
        value: ""
      - key: kubernetes.io/cluster/the-cluster-name
        value: shared
```

assuming we allow a TagSelector to match zero subnets.

This seems overly complicated to me. It's much simpler to rely on the tags being provisioned appropriately:

```yaml
spec:
    subnets:
    - tagKey: ingressclass/the-cluster-name
      tagValue: the-ingress-class
```